test/test/cpu/CPUTimeService.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package test.cpu;
+
+import com.sun.tools.attach.AttachNotSupportedException;
+import com.sun.tools.attach.VirtualMachine;
+
+import javax.management.MBeanServerConnection;
+import javax.management.ObjectName;
+import javax.management.remote.JMXConnector;
+import javax.management.remote.JMXConnectorFactory;
+import javax.management.remote.JMXServiceURL;
+import java.io.IOException;
+import java.util.Set;
+
+/**
+ * Obtain the CPU time of a process using the Attach API
+ */
+class CPUTimeService implements AutoCloseable {
+
+    private final VirtualMachine vm;
+    private final MBeanServerConnection mbeanServer;
+    private final JMXConnector jmxConnector;
+
+    CPUTimeService(long pid) throws IOException, AttachNotSupportedException {
+        vm = VirtualMachine.attach(pid + "");
+        String connectorAddress = vm.startLocalManagementAgent();
+        JMXServiceURL url = new JMXServiceURL(connectorAddress);
+        jmxConnector = JMXConnectorFactory.connect(url);
+        mbeanServer = jmxConnector.getMBeanServerConnection();
+    }
+
+    long getProcessCPUTimeNanos() throws Exception {
+        Set<ObjectName> mbeans = mbeanServer.queryNames(new ObjectName("java.lang:type=OperatingSystem"), null);
```

Comment:
Why queryNames? Can we just call `getAttribute(ObjectName name, String attribute)` for simplicity?

Reply:
This seems to lead to errors, so no.

Reply:
Sorry it's not clear to me. What errors?

Reply:
`getAttribute` is used to get attributes of Means, but to get an MBean, one needs to use `queryNames`.

Reply:
Oh, you're right.

Reply:
`queryNames` returns a set of `ObjectName`s. But you can as well construct the same `ObjectName` directly. What am I missing here? Can you provide an exact exception?

Reply:
I was wrong and fixed it already.